cobigen-maven/cobigen-maven-plugin/src/main/java/com/devonfw/cobigen/maven/GenerateMojo.java

```diff
@@ -223,18 +222,26 @@ private CobiGen createCobiGenInstance() throws MojoExecutionException, MojoFailu
             }
         }
 
-        URL contextXmlUrl = classRealm.getResource("context.xml");
-        if (contextXmlUrl == null) {
-            getLog().error("No context.xml could be found in the classpath!");
-            return null;
+        Path rootPath;
+        URL rootUrl = classRealm.getResource("context.xml");
+        if (rootUrl == null) {
+            rootUrl = classRealm.getResource("src/main/templates/context.xml");
```

Comment:
Just from what I now understood. I am a little bit afraid, that we introduced anoter issue quite some time ago.
I am pretty sure, that copying the template folder to a different place under control, deleting all target and bin folders, and setting this folder as the templateConfigurationFolder in a maven execution, maven will fail generating code in case of any JavaUtil function is invoked from a template.
Can you try it out? 

This is due to the fact, that we rely on an external compiler like eclipse to compile the sources. Such that a maven correct and compliant way would be to call the maven compiler and resource processing on the template folder as well. This would even have resolved the issue with the context.xml as the context.xml will be copied to the target folder at packaging time.

As this feature is not really used by anybody, but it is quite nice for testing, I would like to keep it as is. However, we should be aware of the drawbacks and document them properly. Can you adapt the documentation to that configuration parameter to not let users strugle with this again?